dotnet-client-libraries/ITextBasics/ITextBasics/BigDocument.cs

```diff
@@ -0,0 +1,87 @@
+﻿using iText.IO.Image;
+using iText.Kernel.Pdf;
+using iText.Layout;
+using iText.Layout.Element;
+using iText.Layout.Properties;
+using System.Diagnostics;
+using System.Reflection;
+
+namespace ITextBasics
+{
+    public class BigDocument
+    {
+        public void Create(string pdfFileName)
+        {
+            using var writer = new PdfWriter(pdfFileName);
+            using var pdfDocument = new PdfDocument(writer);
+            using var document = new Document(pdfDocument);
+
+            string? executablePath = Path.GetDirectoryName(Assembly.GetEntryAssembly()!.Location);
+            string imageFile = Path.Combine(executablePath!, "Resources", "Flower.jpg");
+
+            for (int i = 0; i < 3; i++)
+            {
+                document.Add(CreateHeader1(LoremIpsum(3)));
+                document.Add(CreateHeader2(LoremIpsum(7)));
+                document.Add(CreateNormalParagraph(LoremIpsum(100)));
+
+                document.Add(CreateHeader2(LoremIpsum(7)));
+                document.Add(CreateNormalParagraph(LoremIpsum(55)));
+                document.Add(CreateNormalParagraph(LoremIpsum(27)));
+
+                document.Add(CreateHeader1(LoremIpsum(6)));
+                document.Add(CreateHeader2(LoremIpsum(3)));
+                document.Add(CreateNormalParagraph(LoremIpsum(50)));
+                document.Add(CreatePictureParagraph(imageFile));
+                document.Add(CreateHeader2(LoremIpsum(3)));
+                document.Add(CreateNormalParagraph(LoremIpsum(45)));
+            }
+        }
+
+        private Paragraph CreateHeader1(string caption)
+        {
+            return new Paragraph(caption)
+                .SetTextAlignment(TextAlignment.CENTER)
+                .SetFontSize(20)
+                .SetBold();
+        }
+
+        private Paragraph CreateHeader2(string caption)
+        {
+            return new Paragraph(caption)
+                .SetTextAlignment(TextAlignment.LEFT)
+                .SetFontSize(15)
+                .SetBold();
+        }
+
+        private Paragraph CreateNormalParagraph(string text)
+        {
+            return new Paragraph(text);
+        }
+
+        private Paragraph CreatePictureParagraph(string imageFile)
+        {
+            var imageData = ImageDataFactory.Create(imageFile);
+            var image = new Image(imageData);
+
+            return new Paragraph()
+                .Add(image);
+        }
+
+        private string LoremIpsum(int numberOfWords)

```

Comment:
No need to change anything, just an FYI in case you are not aware of the package. There is a faker library called [Bogus](https://github.com/bchavez/Bogus) that includes a Lorem Ipsum generator to make paragraphs, sentences, number of words, etc.

Reply:
Hi,

I Did hear about Bogus library, but I really do not use a lot of external libraries and I would never use an external library for something so simple :)... But thanks to let me know that even something like this exists 👍

Reply:
It is a commonly used library for unit testing and you understand correctly, I was just pointing it out as a "in case you didn't know" statement, no arguments with keeping the method.